src/content/en/updates/posts/2016/03/access-usb-devices-on-the-web.markdown

```diff
@@ -0,0 +1,370 @@
+
+For instance, here's how to get access to a connected Arduino device configured
+to allow the origin.
+
+{% highlight javascript %}
+navigator.usb.requestDevice({ filters: [{ vendorId: 0x2341 }] })
+.then(device => {
+  console.log(device.productName);      // "Arduino Micro"
+  console.log(device.manufacturerName); // "Arduino LLC"
+})
+.catch(error => { console.log(error); });
+{% endhighlight %}
+
+Before you ask, I didn't magically come up with this `0x2341` hexadecimal
+number. I simply searched for the word "Arduino" in this [List of USB
+ID's](http://www.linux-usb.org/usb.ids).
+
+The USB `device` returned in the fulfilled promise above has some basic, yet
+important information about the device such as the supported USB version,
+maximum packet size, vendor and product IDs, the number of possible
+configurations the device can have - basically all fields contained in the
+[device USB Descriptor](http://www.beyondlogic.org/usbnutshell/usb5.shtml#DeviceDescriptors)
+
+For info, if a USB device announces its [support for WebUSB](https://wicg.github.io/webusb/#webusb-platform-capability-descriptor),
+as well as defining a landing page URL, Chrome will show a persistent
+notification when the USB device is plugged in. Clicking on this notification
+will open the landing page.
+
+<img style="width:723px; max-height:345px" src="/web/updates/images/2016-03-02-access-usb-devices-on-the-web/web-usb-notification.png" alt="WebUSB notification"/>
+
+From there, you can simply call `navigator.usb.getDevices` and get access to
+your Arduino device as shown below.
+
+{% highlight javascript %}
+navigator.usb.getDevices().then(devices => {
+  devices.map(device => {
+    console.log(device.productName);      // "Arduino Micro"
+    console.log(device.manufacturerName); // "Arduino LLC"
+  });
+})
+{% endhighlight %}
+
+### Talk to an Arduino USB board
+
+Okay, now let's see how easy it is to communicate from a WebUSB compatible
+Arduino board over the USB port. Check out instructions at
+[https://github.com/webusb/arduino](https://github.com/webusb/arduino) to
+WebUSB-enable your [sketches](http://www.arduino.cc/en/Tutorial/Sketch).
+
+{% highlight javascript %}
+var device;
+
+navigator.usb.requestDevice({ filters: [{ vendorId: 0x2341 }] })
+.then(selectedDevice => {
+   device = selectedDevice;
+   return device.open(); // Begin a session.
+ })
+.then(() => device.selectConfiguration(1)) // Select configuration #1 for the device.
+.then(() => device.claimInterface(2)) // Request exclusive control over interface #2.
+.then(() => device.controlTransferOut({
+    requestType: 'class',
+    recipient: 'interface',
+    request: 0x22,
+    value: 0x01,
+    index: 0x02})) // Ready to receive data
+.then(() => device.transferIn(5, 64)) // Waiting for 64 bytes of data from endpoint #5.
+.then(result => {
```

Comment:
I just found where you describe those. Find some place to mention that you'll cover these calls later in the article.